libfdcore/peers.c

```diff
@@ -197,10 +200,16 @@ int fd_peer_remove ( DiamId_t diamid, size_t diamidlen )
 			peer->p_hdr.info.config.pic_flags.exp = PI_EXP_INACTIVE;
 			peer->p_hdr.info.config.pic_lft = 0;
 			CHECK_FCT( fd_p_expi_update(peer) );
+			found = 1;
 			break;
 		}
 	}
 
+	if (!found)
+	{
+		TRACE_ERROR("Diameter peer %*s not valid - caller is out of sync", (int)diamidlen, diamid);
```

Comment:
I think this is probably a valid path to fall in to (if a peer disappears while we're deleting it).  Also, errors on deleting a non-existent object are often misleading (there's no "error" here from an external point of view) and I generally avoid them.

The log at the beginning of the function gets a :+1 from me though!